contracts/upgradeable_contracts/OverdrawManagement.sol

```diff
@@ -4,28 +4,38 @@ import "../upgradeability/EternalStorage.sol";
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "./Upgradeable.sol";
 import "./RewardableBridge.sol";
+import "./BasicBridge.sol";
 
-contract OverdrawManagement is EternalStorage, RewardableBridge, Upgradeable {
+contract OverdrawManagement is EternalStorage, RewardableBridge, Upgradeable, BasicBridge {
     using SafeMath for uint256;
 
     event UserRequestForSignature(address recipient, uint256 value);
 
-    function fixAssetsAboveLimits(bytes32 txHash, bool unlockOnForeign) external onlyIfUpgradeabilityOwner {
+    function fixAssetsAboveLimits(bytes32 txHash, bool unlockOnForeign, uint256 valueToUnlock)
+        external
+        onlyIfUpgradeabilityOwner
+    {
         require(!fixedAssets(txHash));
+        require(valueToUnlock <= maxPerTx());
         address recipient;
         uint256 value;
         (recipient, value) = txAboveLimits(txHash);
-        require(recipient != address(0) && value > 0);
-        setOutOfLimitAmount(outOfLimitAmount().sub(value));
+        require(recipient != address(0) && value > 0 && value >= valueToUnlock);
+        setOutOfLimitAmount(outOfLimitAmount().sub(valueToUnlock));
+        uint256 pendingValue = value.sub(valueToUnlock);
+        setTxAboveLimitsValue(pendingValue, txHash);
```

Comment:
Let’s also emit a new event to inform the hash, the unlocked value and remaining part

Reply:
Added event https://github.com/poanetwork/poa-bridge-contracts/pull/247/commits/c38f023182894b1bcb6b10f4a363beabeba39e27